### ISSUE TITLE
Failing to prebuild dockerfile: "invalid reference format"

### ISSUE THREAD

**Issue by instagibbs** [cirruslabs/cirrus-cli#256]
Cirrus CI jobs are completing for https://github.com/bitcoin-core/HWI/blob/master/.cirrus.yml, but I'm getting stuck running locally:

```
cirrus run
🕕 'Prebuild ci/cirrus.Dockerfile' Task 0.6s
   Pulling image gcr.io/cirrus-ci-community/gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:latest:latest...
   Image gcr.io/cirrus-ci-community/gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:latest:latest is not available locally nor remotely, building it...

Error: Error response from daemon: invalid reference format
2021/01/25 13:40:45 Error response from daemon: invalid reference format
```

Ideas?

**Comment by edigaryev** [comment]
Hello Gregory, thanks for your report!

This should be now fixed as a part of [0.30.5 release](https://github.com/cirruslabs/cirrus-cli/releases/tag/v0.30.5), the binaries and the new Homebrew formula will be available shortly.

**Comment by instagibbs** [comment]
efficient, thank you!

**Comment by instagibbs** [comment]
Apologies, but now I'm getting:

```
cirrus run
✅ 'Prebuild ci/cirrus.Dockerfile' Task 25s
🕑 'lint' task 1s
   ✅ Preparing execution environment... 1.3s
Error: Error response from daemon: No such image: gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:latest
2021/01/26 09:54:32 Error response from daemon: No such image: gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:latest
```

**Comment by edigaryev** [comment]
>`Error response from daemon: No such image:`

Hmm, it looks that the image built in the `Prebuild ci/cirrus.Dockerfile` step has a different name than `gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:latest` for some reason or it was deleted before the actual task has started.

Could you please re-run the build using the `cirrus run -o simple -v` command?

**Comment by instagibbs** [comment]
```
cirrus run -o simple -v
gRPC server is listening at http://127.0.0.1:63931 (http://host.docker.internal:63931 inside of a container)
running task Prebuild ci/cirrus.Dockerfile (48)
Started 'Prebuild ci/cirrus.Dockerfile' Task
Pulling image gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:latest...
Image gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:latest is not available locally nor remotely, building it...
Step 1/5 : FROM ubuntu:trusty
---> df043b4f0cf1
Step 2/5 : ENV DEBIAN_FRONTEND=noninteractive
---> Using cache
---> 2285779eaac3
Step 3/5 : RUN apt-get update
---> Using cache
---> 531af028d436
Step 4/5 : RUN apt-get install -y     build-essential     autotools-dev     automake     cmake     pkg-config     bsdmainutils     libtool     curl     git     ccache     qemu-user-static     libsdl2-dev     libsdl2-image-dev     gcc-arm-none-eabi     libnewlib-arm-none-eabi     gcc-arm-linux-gnueabihf     libc6-dev-armhf-cross     libudev-dev     libusb-1.0-0-dev     libssl-dev     libevent-dev     libdb-dev     libdb++-dev     libboost-system-dev     libboost-filesystem-dev     libboost-chrono-dev     libboost-test-dev     libboost-thread-dev     protobuf-compiler     cython3
---> Running in b78b14c4b880
Reading package lists...
Building dependency tree...
Reading state information...
Some packages could not be installed. This may mean that you have
requested an impossible situation or if you are using the unstable
distribution that some required packages have not yet been created
or been moved out of Incoming.
The following information may help to resolve the situation:
The following packages have unmet dependencies:
libsdl2-dev : Depends: libsdl2-2.0-0 (= 2.0.2+dfsg1-3ubuntu1.3) but it is not going to be installed
               Depends: libgles2-mesa-dev
               Depends: libglu1-mesa-dev but it is not going to be installed
 libsdl2-image-dev : Depends: libsdl2-image-2.0-0 (= 2.0.0+dfsg-3+deb8u1build0.14.04.1) but it is not going to be installed
E: Unable to correct problems, you have held broken packages.

while closing temporary archive file: close /var/folders/z_/5934r9jn09s8ld5trxt6r1hw0000gn/T/cirrus-prebuilt-archive-678003830: file already closed
task Prebuild ci/cirrus.Dockerfile (48) succeeded
'Prebuild ci/cirrus.Dockerfile' Task succeeded in 4.5s!
running task lint (0)
Started 'lint' task
Started 'Preparing execution environment...'
Preparing volume to work with...
'Preparing execution environment...' succeeded in 1.4s!
creating container using working volume cirrus-working-volume-31f946c8-636b-4cf9-9679-4c4b6a5dcdb2
Error: Error response from daemon: No such image: gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:latest
2021/01/27 09:30:20 Error response from daemon: No such image: gcr.io/cirrus-ci-community/d41d8cd98f00b204e9800998ecf8427e:lates
```

**Comment by edigaryev** [comment]
>E: Unable to correct problems, you have held broken packages.

Looks like you're using an old base image that is not updated automatically by the CLI due to recently introduced [Docker Hub rate limits](https://www.docker.com/increase-rate-limits).

You can generally use the `--container-pull` option to work around that, but I've found a few more issues in the log that you've provided that prevent this, and there's a fix available in #261, so you can either wait for the next release or manually run:

```
docker pull ubuntu:trusty
```

If that doesn't work it's quite possible that the `ubuntu:trusty` is simply not supported anymore and you're on your own. LTS support for that release [seems to be dropped in 2019](https://en.wikipedia.org/wiki/Ubuntu_version_history#Ubuntu_14.04_LTS_(Trusty_Tahr)).

**Comment by edigaryev** [comment]
> there's a fix available in #261

Now available in the latest [0.31.0 release](https://github.com/cirruslabs/cirrus-cli/releases/tag/v0.31.0).
